linux/shim/v1/shim.proto

```diff
@@ -9,53 +9,67 @@ import "github.com/containerd/containerd/api/types/mount.proto";
 import "github.com/containerd/containerd/api/types/task/task.proto";
 import "google/protobuf/timestamp.proto";
 
-option go_package = "github.com/containerd/containerd/api/services/shim/v1;shim";
+option go_package = "github.com/containerd/containerd/linux/shim/v1;shim";
 
 // Shim service is launched for each container and is responsible for owning the IO
 // for the container and its additional processes.  The shim is also the parent of
 // each container and allows reattaching to the IO and receiving the exit status
 // for the container processes.
 service Shim {
-	rpc Create(CreateRequest) returns (CreateResponse);
-	rpc Start(StartRequest) returns (google.protobuf.Empty);
-	rpc Delete(DeleteRequest) returns (DeleteResponse);
-	rpc DeleteProcess(DeleteProcessRequest) returns (DeleteResponse);
+	// State returns shim and task state information.
 	rpc State(StateRequest) returns (StateResponse);
-	rpc Processes(ProcessesRequest) returns (ProcessesResponse);
-	rpc Pause(PauseRequest) returns (google.protobuf.Empty);
-	rpc Resume(ResumeRequest) returns (google.protobuf.Empty);
-	rpc Checkpoint(CheckpointRequest) returns (google.protobuf.Empty);
 
-	rpc Exit(ExitRequest) returns (google.protobuf.Empty);
-	rpc Events(EventsRequest) returns (stream containerd.v1.types.Event);
+	rpc Create(CreateTaskRequest) returns (CreateTaskResponse);
+
+	rpc Start(StartTaskRequest) returns (google.protobuf.Empty);
+
+	rpc Delete(DeleteTaskRequest) returns (DeleteResponse);
+
+	rpc DeleteProcess(DeleteProcessRequest) returns (DeleteResponse);
+
+	rpc ListProcesses(ListProcessesRequest) returns (ListProcessesResponse);
+
+	rpc Pause(PauseTaskRequest) returns (google.protobuf.Empty);
+
+	rpc Resume(ResumeTaskRequest) returns (google.protobuf.Empty);
+
+	rpc Checkpoint(CheckpointTaskRequest) returns (google.protobuf.Empty);
+
+	rpc Stream(StreamEventsRequest) returns (stream containerd.v1.types.Event);
```

Comment:
Let's move this event type out of the types package. Presumably, this is not part of the regular events subsystem?